reference/batch-workflow-syntax.md

```diff
@@ -189,6 +204,172 @@ cache:
 
 **Expression contexts:** [`flow` context](batch-contexts.md#flow-context), [`params` context](batch-contexts.md#params-context), [`env` context](batch-contexts.md#env-context), [`tags` context](batch-contexts.md#tags-context), [`volumes` context](batch-contexts.md#volumes-context), [`images` context](batch-contexts.md#images-context).
 
+## `images`
+
+A mapping of image definitions used by the _batch_ workflow.
+
+If the specified image reference is available at the Neu.ro registry and the [`.force_rebuild`](batch-workflow-syntax.md#images-less-than-image-id-greater-than-force-rebuild) flag is not set, then Neu.ro Flow starts building the image from scratch. 
+If this flag is set, then the existing image from the registry will be used.
+
+{% hint style="info" %}
+The `images` section is not required. A task can specify the image name in a plain string without referring to the `${{ images.my_image.ref }}` context.
+
+However, this section exists for convenience: there is no need to repeat yourself if you can just point the image reference everywhere in the YAML.
+{% endhint %}
+
+### `images.<image-id>`
+
+The key `image-id` is a string and its value is a map of a bake's configuration data. You must replace `<image-id>` with a string that is unique to the `images` object. `<image-id>` must start with a letter and contain only alphanumeric characters or underscore symbols `_`. Dash `-` is not allowed.
+
+### `images.<image-id>.ref`
+
+**Required** Image _reference_ that can be used in the [`tasks.image`](batch-workflow-syntax.md#tasks-image) expression.
+
+**Example of a self-hosted image:**
+
+```yaml
+images:
+  my_image:
+    ref: image:my_image:latest
+```
+
+This can only use locally accessible functions (such as `hash_files`). Its value will be calculated before the remote executor starts.
+
+**Example of external image:**
+
+```yaml
+images:
+  python:
+    ref: python:3.9.0
+```
+
+{% hint style="info" %}
+Use the embedded [`hash_files()`](expression-functions.md#hash_files-pattern) function to generate the built image's tag based on its content.
+{% endhint %}
+
+**Example of an auto-calculated stable hash:**
+
+```yaml
+images:
+  my_image:
+    ref: image:my_image:${{ hash_files('Dockerfile', 'requirements/*.txt', 'modules/**/*.py') }}
+```
+
+**Expression contexts:** [`flow` context](batch-contexts.md#flow-context).
+
+### `images.<image-id>.context`
+
+The Docker _context_ used to build an image. Can be either local path (e.g. ${{ flow.workspace / 'some/dir' }}) or a remote volume spec (e.g. storage:subdir/${{ inputs.path_on_storage }}). 
```

Comment:
`inputs` context is only available in actions, so it can be confusing to use it in example for batch workflow.
```suggestion
The Docker _context_ used to build an image. Can be either local path (e.g. ${{ flow.workspace / 'some/dir' }}) or a remote volume spec (e.g. storage:subdir/${{ flow.flow_id }}). 
```